sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationFilter.java

```diff
@@ -106,14 +114,14 @@ protected void doFilterInternal(HttpServletRequest httpServletRequest,
             UserPrincipal userPrincipal = (UserPrincipal) httpSession.getAttribute(CURRENT_USER_PRINCIPAL);
             if (userPrincipal == null
                 || !userPrincipal.getAadIssuedBearerToken().equals(aadIssuedBearerToken)
-                || userPrincipal.getAccessTokenForGraphApi() == null
             ) {
                 userPrincipal = userPrincipalManager.buildUserPrincipal(aadIssuedBearerToken);
-                String tenantId = userPrincipal.getClaim(AADTokenClaim.TID).toString();
-                String accessTokenForGraphApi = azureADGraphClient
-                    .acquireTokenForGraphApi(aadIssuedBearerToken, tenantId)
-                    .accessToken();
-                userPrincipal.setAccessTokenForGraphApi(accessTokenForGraphApi);
+                String accessTokenForGraphApi = accessTokenManager
```

Comment:
In the case of implicit flow, users don't configure tenantId as properties, so we should acquire tenantId from UserPrincipal and pass it to AccessTokenManager to acquire token.